RestaurantReservation.API/Controllers/CustomersController.cs

```diff
@@ -0,0 +1,144 @@
+public class CustomersController : ControllerBase
+{
+    private readonly ICustomerRepository _customerRepository;
+    private readonly IMapper _mapper;
+
+    public CustomersController(ICustomerRepository customerRepository, IMapper mapper)
+    {
+        _customerRepository = customerRepository;
+        _mapper = mapper;
+    }
+
+    /// <summary>
+    /// Gets all customers.
+    /// </summary>
+    /// <returns>A list of customer data transfer objects.</returns>
+    /// <response code="200">Returns the list of customers.</response>
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IEnumerable<CustomerDto>))]
+    public async Task<ActionResult<IEnumerable<CustomerDto>>> GetCustomers()
+    {
+        var customers = await _customerRepository.GetAllCustomersAsync();
+        return Ok(_mapper.Map<IEnumerable<CustomerDto>>(customers));
+    }
+
+    /// <summary>
+    /// Gets a specific customer by ID.
+    /// </summary>
+    /// <param name="customerId">The ID of the customer to retrieve.</param>
+    /// <param name="includeReservations">Flag to include reservations in the response.</param>
+    /// <returns>The customer data transfer object.</returns>
+    /// <response code="200">Returns the requested customer.</response>
+    /// <response code="404">If the customer with the specified ID is not found.</response>
+    [HttpGet("{customerId}", Name = "GetCustomer")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(CustomerDto))]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<CustomerDto>> GetCustomer(int customerId, bool includeReservations = false)
+    {
+        var customer = await _customerRepository.GetCustomerAsync(customerId, includeReservations);
+        if (customer == null)
+        {
+            return NotFound(ApiErrors.CustomerNotFound);
+        }
+
+        return includeReservations
+            ? Ok(_mapper.Map<CustomerWithReservationsDto>(customer))
+            : Ok(_mapper.Map<CustomerDto>(customer));
+    }
+
+    /// <summary>
+    /// Creates a new customer.
+    /// </summary>
+    /// <param name="customer">The data for the new customer.</param>
+    /// <returns>The created customer data transfer object.</returns>
+    /// <response code="201">Returns the created customer.</response>
+    /// <response code="400">If data is invalid.</response>
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status201Created, Type = typeof(CustomerDto))]
+    public async Task<ActionResult<CustomerDto>> PostCustomer(CustomerCreationDto customer)
+    {
+        var customerEntity = _mapper.Map<Customer>(customer);
+        await _customerRepository.CreateCustomerAsync(customerEntity);
+        await _customerRepository.SaveChangesAsync();
+
+        return CreatedAtRoute("GetCustomer",
+            new { 
+                customerId = customerEntity.CustomerId 
+            },_mapper.Map<CustomerDto>(customerEntity));
```

Comment:
to adhere to best practices, you should introduce a service layer to abstract this logic. The service would:

- handle  mapping
- orchestrates and interact with repository calls
- encapsulate all domain logic